### PR TITLE
kubeadm: If `--config` is set, don't allow any other option as it won't have effect

### PR DESCRIPTION
**Pull request by xilabao**
If use config in kubeadm init, cann't get other values from other arguments. 
`kubeadm init --config=../kubeadm.config --token 447ad3.96cda76e3206fca0 --apiserver-bind-port 6445`

So I think we need to allow get values from command and is prior than cofig file.

### CODE REVIEW

**Comment by mikedanese** [comment]
I think we want file to be used exclusively if it is passed. Can you make this error if a mix is used?

**Comment by xilabao** [comment]
@mikedanese add a warning in the help? If we make this error, we need to check every option too.  BTW, `kubectl` support to use --kubeconfig and --certificate-authority (or others) together.  do we really don't want this?

**Comment by xilabao** [comment]
ping @luxas

**Comment by xilabao** [comment]
I have changed the patch as you expect. @mikedanese PTAL

**Comment by luxas** [comment]
/lgtm

Waiting for @mikedanese for approval

@k8s-bot ok to test

**Comment by mikedanese** [comment]
I very much disagree with this change. Please hold off til sig meeting where I will be presenting a doc on configuring kubernetes components.

**Comment by mikedanese** [comment]
We should not support mixed flags. If you want to validate that if --config is passed, we only pass one arg, I would be ok with that.

**Comment by luxas** [comment]
> If you want to validate that if --config is passed, we only pass one arg, I would be ok with that.

@xilabao please do that instead

**Comment by xilabao** [comment]
@mikedanese @luxas PTAL

**Review by luxas — APPROVED** [review]
LGTM, but I defer to @mikedanese

**Comment by xilabao** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this
@k8s-bot pull-kubernetes-e2e-kops-aws test this

**Comment by luxas** [comment]
/lgtm

This should be a cherrypick-candidate for v1.7 (cc @caesarxuchao), somehow we missed to get this in a long time ago.

It's a very small patch;  ~10 LoC, but with ~120 LoC of testing :+1:

**Review by luxas — APPROVED** [review]

**Comment by caesarxuchao** [comment]
Approving the cherrypick. Depending on if this PR is merged before my bactch cherrypick today, you might need to create a manual cherrypick.

**Comment by luxas** [comment]
@caesarxuchao It's not critical, so I can create the cherrypick myself after I know everything's alright and it has merged.

**Pull request merged**
